src/post/workshed_utils.ts

```diff
@@ -67,6 +67,26 @@ export function getBestCycle(): TrainSet.Cycle {
   return [...trainCycle];
 }
 
+export function getBestOffset(): number {
+  const cycle = getBestCycle();
+  const valueTrainStation = (station: TrainSet.Station) =>
+    POTENTIAL_BEST_TRAIN_PIECES.find(({ piece }) => piece === station)?.value?.() ?? 0;
+  const cycleValue =
+    5 * (sum(POTENTIAL_BEST_TRAIN_PIECES, ({ value }) => value()) + valueTrainStation(cycle[0]));
+
+  const potentialOffsets = Array(POTENTIAL_BEST_TRAIN_PIECES.length)
+    .fill(null)
+    .map((_x, i) => i + 2);
```

Comment:
We should only need to consider offsets [2,3,4,5,6,7], which would only require `POTENTIAL_BEST_TRAIN_PIECES.length - 2` (after all, we are already using the magic number `2` in `i + 2`)